### DISCUSSION TITLE
How to call specific attributes of another class

### DISCUSSION THREAD

**Discussion by BENFREDJRima-34** [mesa/mesa#1551] [Q&A] [answered]
Hi, I have two classes state and MyAgent
I need to call state attributes in MyAgent class like this
```
class state (Enum): 
    R_advantage = random.uniform(0,1)
    Copl = random.uniform(0,1)
    Copa = random.uniform(0,1)
    Observ= 1

class MyAgent(Agent):
    """An agent with fixed initial wealth."""
    def __init__(self, unique_id, model):
        super().__init__(unique_id, model)
        self.unique_id = unique_id
        self.state1 = 0
        self.RA = self.state.R_advantage
        self.Acc = random.uniform(0,1)
        self.Uncer = self.state.Copl
        self.Infl = 0 `
```

AttributeError: 'MyAgent' object has no attribute 'state'

**Comment by rht** [comment, marked as answer]
```
self.RA = state.R_advantage
self.Uncer = state.Copl
```

**Reply by BENFREDJRima-34** [reply]
if I put it like this in the comparison function
I will get the below error
`def comparison(self):
    if (self.Copa) >= random.uniform(0.75,1) and (self.Acc > self.Uncer):
               self.state1= 1`
TypeError: '>=' not supported between instances of 'state' and 'float'

**Reply by rht** [reply]
Remove `Enum` in
`class state (Enum): `

**Reply by BENFREDJRima-34** [reply]
It's okay, I removed it, Thank you so much
